cobra/core/Model.py

```diff
@@ -230,9 +308,85 @@ def add_reactions(self, reaction_list):
 
         self.reactions += reaction_list
 
+        # from cameo ...
+        self._populate_solver(reaction_list)
+
+    def _populate_solver(self, reaction_list, metabolite_list=None):
+        """Populate attached solver with constraints and variables that
+        model the provided reactions.
+        """
+        constraint_terms = AutoVivification()
+        if metabolite_list is not None:
```

Comment:
The metabolite cannot be added to a reaction using the `reaction.add_metabolites` method if it doesn't have a constraint. Having metabolites that don't participate in reactions is not that common, and the extra work for the solver is pretty close to nothing, so the most straightforward way to handle it was to just make sure that all metabolites have a corresponding constraint.